### ISSUE TITLE
Regexes for BugFixMessage are too leniant

### ISSUE THREAD

**Issue by cflewis** [SoftwareIntrospectionLab/MininGit#41]
The regexes should be changed to only match when a letter or symbol is not in front of them, so something like "debug" doesn't match, but "This is a sentence. Bugs were found here" does.

**Comment by supertri** [comment]
1) Referenced file name contains "debug", "fix", etc.
2) Referenced method name contains "debug", "fix", etc.
3) Referenced macro or field contains "DEBUG"
4) There was one commit id that talked about a format change so that
"ff== bugfix"
5) Compound words: "prefix" or "errorlog"
6) "Patch" may not refer to a bug fix.

**Comment by cflewis** [comment]
From email from supertri:

```
I think you should:

1) Change the regexp so that it requires whitespace surrounding the
keywords, but also add plural versions of the keywords, and be sure it
includes combinations like "bugfix". This should deal with 1, 2, and 5
from above. It will probably deal with most instances of 3 as well.
2) Not worry about case 4. This is a false positive, but it seems
pretty uncommon, and I am not sure how to identify these types of
cases (ideas?). Or we could filter out "== bugfix", and then scan log
messages for specific weird strings (but that feels a bit hacky).
3) I am undecided on what to do about point 3. Should we keep "patch"
as a keyword or drop it? It could lead to some false positives, but
those seem less obviously false than cases 1-3 and 5. Dropping it
could lead to false negatives. I think this is a judgement call.

Once the regexp has been updated, add a second "is_bug_fix_tight"
boolean in the DB with the new values. Take a look at a few random
samples where there is a difference between the two booleans to make
sure there is nothing obvious we are missing.
```

**Comment by cflewis** [comment]
Fixed in 99dd394ce8e942c66f031ad9dbc4c48421b9d481
